### PR TITLE
tfmini cleanup and use PX4Rangefinder

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

File: src/drivers/distance_sensor/tfmini/TFMINI.cpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+int
+TFMINI::init()
+{
+	int32_t hw_model = 1; // only one model so far...
+
+	switch (hw_model) {
+	case 1: // TFMINI (12m, 100 Hz)
+		_px4_rangefinder.set_min_distance(0.3f);
+		_px4_rangefinder.set_max_distance(12.0f);
+		_px4_rangefinder.set_fov(math::radians(1.15));
```
Comment (mcsauder) [inline src/drivers/distance_sensor/tfmini/TFMINI.cpp]: Hi @dagar , would you want to set these numbers as #defines in the header file?

Everything else looks great.  Bench tests are all functioning as expected.

Comment (dagar) [reply src/drivers/distance_sensor/tfmini/TFMINI.cpp]: Personally I don't find this simple self explanatory use of "magic numbers" offensive, but I'm interested if there's a compelling reason I'm wrong.

Comment (dagar) [inline src/drivers/distance_sensor/tfmini/TFMINI.cpp]:
```suggestion
		_px4_rangefinder.set_fov(math::radians(1.15f));
```

File: src/drivers/distance_sensor/tfmini/TFMINI.cpp [hunk 2]
```diff
@@ -198,7 +200,7 @@ TFMINI::collect()
 
 		// parse buffer
 		for (int i = 0; i < ret; i++) {
-			tfmini_parse(readbuf[i], _linebuf, &_linebuf_index, &_parse_state, &distance_m);
+			ret_parse = tfmini_parse(readbuf[i], _linebuf, &_linebuf_index, &_parse_state, &distance_m);
```
Comment (dagar) [inline src/drivers/distance_sensor/tfmini/TFMINI.cpp]: TODO: review if this is correct
We want to only publish on successful parse of new valid distance.

File: src/drivers/distance_sensor/tfmini/TFMINI.cpp [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+		}
+
+		// bytes left to parse
+		bytes_available -= ret;
+
+	} while (bytes_available > 0);
+
+	// no valid measurement after parsing buffer
+	if (distance_m < 0.0f) {
```
Comment (mhkabir) [inline src/drivers/distance_sensor/tfmini/TFMINI.cpp]: @dagar this was doing the equivalent of the ret_parse you just added.

Comment (dagar) [reply src/drivers/distance_sensor/tfmini/TFMINI.cpp]: Ah, thanks. I'll remove it.

**Pull request merged**